crypto/crypto_test.go

```diff
@@ -1,88 +1,60 @@
-// Copyright 2014 The go-ethereum Authors
-// This file is part of the go-ethereum library.
-//
-// The go-ethereum library is free software: you can redistribute it and/or modify
-// it under the terms of the GNU Lesser General Public License as published by
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-//
-// The go-ethereum library is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
-// GNU Lesser General Public License for more details.
-//
-// You should have received a copy of the GNU Lesser General Public License
-// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
-
-// THIS WAS TAKEN FROM github.com/ethereum/go-ethereum MODIFY CAREFULLY
-
-package crypto
+package crypto_test
 
 import (
-	"crypto/ecdsa"
 	"testing"
 
-	"gx/ipfs/QmZp3eKdYQHHAneECmeK6HhiMwTPufmjC8DuuaGKv3unvx/blake2b-simd"
-
 	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/assert"
-	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/require"
-)
 
-func TestToECDSAErrors(t *testing.T) {
-	assert := assert.New(t)
-
-	_, err := HexToECDSA("0000000000000000000000000000000000000000000000000000000000000000")
-	assert.Error(err)
-
-	_, err = HexToECDSA("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff")
-	assert.Error(err)
-}
+	"github.com/filecoin-project/go-filecoin/crypto"
+)
 
-func TestSign(t *testing.T) {
+func TestGenerateKey(t *testing.T) {
 	assert := assert.New(t)
-	require := require.New(t)
 
-	// Generate a key pair and convert the public key to bytes for easy comparison
-	key, err := GenerateKey()
+	sk, err := crypto.GenerateKey()
 	assert.NoError(err)
-	ecdasKey, ok := key.Public().(*ecdsa.PublicKey)
-	require.True(ok)
-	// This is the key we will compare recovered keys against
-	expPub := ECDSAPubToBytes(ecdasKey)
 
-	// Sign a message using the key we generated previously
-	msg := blake2b.Sum256([]byte("have you considered not asking passive aggressive questions?"))
-	sig, err := Sign(msg[:], key)
-	assert.NoError(err)
-
-	// recover a public key from a message and a signature, it should be
-	// equal to the previously generated public key
-	recPub, err := Ecrecover(msg[:], sig)
-	assert.NoError(err)
-	assert.Equal(expPub, recPub)
+	assert.Equal(len(sk), 32)
+	assert.NotEqual(sk[0], 0)
```

Comment:
why not zero? please add a comment for the unfamiliar whose reaction to seeing this line like is that it is going to flake 1/256 runs.